File: R project/microproj1.r
mat <- matrix(c(-3,3, 5,-5), nrow = 2, ncol = 2, byrow = TRUE)
prec = 0.01

valx <- c()
val1 <- c()
val2 <- c()
val3 <- c()
val4 <- c()

for (i in (0:100))
{
    temp <- Matrix::expm(mat*i*prec)
    val1 <- c(val1, temp[1,1])
    val2 <- c(val2, temp[1,2])
    val3 <- c(val3, temp[2,1])
    val4 <- c(val4, temp[2,2])
    valx <- c(valx, i*prec)
}

jpeg("Rplot.jpg")
#Create the graph
plot(valx, val1, "l", main = "Exponential of a matrix", col = 'blue', xlab = "X values", ylab = "Y values", ylim = c(0,1))
lines(valx, val2, col = 'blue')
lines(valx, val3, col = 'red')
lines(valx, val4, col = 'red')

#Add the grid
abline(v = 0:173,  lty = 2, col = "grey")
abline(h = 0:5,  lty = 2, col = "grey")

#Resize the graph
options(repr.plot.width=15, repr.plot.height=8)

File: Python project/microproj1.py
import numpy as np
import matplotlib.pyplot as plt
from scipy.linalg import expm

matrix = np.mat([[-3,3],[5,-5]])
prec = 0.01

exp = [expm(matrix * i * prec) for i in range(100)]
valx = [i * prec for i in range(100)]

val1, val2, val3, val4 = [], [], [], []
for mat in exp:
    val1.append(mat[0][0])
    val2.append(mat[0][1])
    val3.append(mat[1][0])
    val4.append(mat[1][1])
    
#Create the graph
plt.plot(valx, val1, linewidth = 1, color = 'blue')
plt.plot(valx, val2, linewidth = 1, color = 'blue')
plt.plot(valx, val3, linewidth = 1, color = 'red')
plt.plot(valx, val4, linewidth = 1, color = 'red')

#Resize the figure
fig_size = plt.rcParams["figure.figsize"]
fig_size[0] = 15
fig_size[1] = 5

#Add the title and the axis caption
plt.title('Exponential of a matrix')
plt.ylabel('Y value')
plt.xlabel('X value')

#Add the grid
plt.grid(True)

plt.savefig("Pplots.pdf")
plt.savefig("Pplots.png")